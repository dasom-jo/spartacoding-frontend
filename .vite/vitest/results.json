{"version":"2.1.5","results":[[":week-1/src/Snapshot.test.tsx",{"duration":4.370999999999981,"failed":true}]]}