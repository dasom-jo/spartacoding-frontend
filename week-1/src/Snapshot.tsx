import { useState } from 'react';

const Snapshot = () => {
  const books: { [key: string]: string } = {
    javascript: '2층 143번',
    react: '3층 432번',
    typescript: '4층 242번',
  };
  //사용자가 검색한 책 제목 상태 관리
  const [bookName, setBookName] = useState('');
  //해당 값 (책의 존재 유무) 상태 관리
  const [result, setResult] = useState('');

  //검색버튼 클릭시 이벤트
  const handleSearch = () => {
    if (bookName in books) {
      setResult(`${bookName}의 위치는 ${books[bookName]}에 있습니다.`);
    } else {
      setResult(`${bookName}을 찾을 수 없습니다.`);
    }
  };

  return (
    <>
      <input
        placeholder="책을 검색해주세요"
        value={bookName}
        onChange={(e) => setBookName(e.target.value)}
      />
      <button onClick={handleSearch}>검색</button>
      <div>{result}</div>
    </>
  );
};

export default Snapshot;
