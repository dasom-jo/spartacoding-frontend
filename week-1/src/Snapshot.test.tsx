import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Snapshot from "./Snapshot";

describe("검색 컴포넌트 스냅샷 테스트", () => {
  it("초기 렌더링 상태를 스냅샷으로 저장", () => {
    const { container } = render(<Snapshot />);
    expect(container).toMatchSnapshot(); // 초기 렌더링 스냅샷 저장
  });

  it("검색어 입력 후 렌더링 상태를 스냅샷으로 저장", () => {
    const { container } = render(<Snapshot />);
    const input = screen.getByPlaceholderText("책을 검색해주세요");

    fireEvent.change(input, { target: { value: "리액트" } });

    expect(container).toMatchSnapshot(); // 검색어 입력 후 UI 스냅샷 저장
  });

  it("검색 버튼 클릭 후 결과가 표시된 상태를 스냅샷으로 저장", async () => {
    const { container } = render(<Snapshot />);
    const input = screen.getByPlaceholderText("책을 검색해주세요");
    const button = screen.getByText("검색");

    fireEvent.change(input, { target: { value: "리액트" } });
    fireEvent.click(button);

    await waitFor(() => {
      expect(container).toMatchSnapshot(); // 검색 결과 표시 후 UI 스냅샷 저장
    });
  });
});
