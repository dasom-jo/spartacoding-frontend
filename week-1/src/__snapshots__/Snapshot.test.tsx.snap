// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`검색 컴포넌트 스냅샷 테스트 > 검색 버튼 클릭 후 결과가 표시된 상태를 스냅샷으로 저장 1`] = `
<div>
  <input
    placeholder="책을 검색해주세요"
    value="리액트"
  />
  <button>
    검색
  </button>
  <div>
    리액트을 찾을 수 없습니다.
  </div>
</div>
`;

exports[`검색 컴포넌트 스냅샷 테스트 > 검색 버튼 클릭 후 결과가 표시된 상태를 스냅샷으로 저장 2`] = `
<div>
  <input
    placeholder="책을 검색해주세요"
    value="리액트"
  />
  <button
    style="background-color: red;"
  >
    검색
  </button>
  <div>
    리액트을 찾을 수 없습니다.
  </div>
</div>
`;

exports[`검색 컴포넌트 스냅샷 테스트 > 검색어 입력 후 렌더링 상태를 스냅샷으로 저장 1`] = `
<div>
  <input
    placeholder="책을 검색해주세요"
    value="리액트"
  />
  <button>
    검색
  </button>
  <div />
</div>
`;

exports[`검색 컴포넌트 스냅샷 테스트 > 초기 렌더링 상태를 스냅샷으로 저장 1`] = `
<div>
  <input
    placeholder="책을 검색해주세요"
    value=""
  />
  <button>
    검색
  </button>
  <div />
</div>
`;
